util/patch/patch.go

```diff
@@ -146,10 +146,19 @@ func (h *Helper) Patch(ctx context.Context, obj client.Object, opts ...Option) e
 	if err := h.patch(ctx, obj); err != nil {
 		errs = append(errs, err)
 	}
+
 	if err := h.patchStatus(ctx, obj); err != nil {
 		errs = append(errs, err)
 	}
 
+	if obj.GetDeletionTimestamp() != nil && len(obj.GetFinalizers()) == 0 && len(errs) > 0 {
+		for _, err := range errs {
+			if ok := apierrors.IsNotFound(err); ok {
+				return nil
+			}
+		}
+	}
+
```

Comment:
@sbueringer not sure if this is the check you wanted from the discussion but here we are 
1. Trying to patch the status
2. Checking to see after the patching of the status if we have a deletion time stamp, 0 finalizers, and if the error in the error(s) are `IsNotFound` to ignore them

Reply:
I would suggest this instead
```go
	if err := h.patchStatus(ctx, obj); err != nil {
		if !(apierrors.IsNotFound(err) && !obj.GetDeletionTimestamp().IsZero() && len(obj.GetFinalizers()) == 0) {
			errs = append(errs, err)
		}
	}
```

The idea is that we precisely only ignore not found errors coming back from patchStatus

Reply:
This makes sense, resolving as this is updated on the latest push.